crates/orchestrator/src/lib.rs

```diff
@@ -482,9 +480,9 @@ where
                 // Deal with the bench results from different nodes
                 let cur_metrics = self.bench_results.clone();
                 self.bench_results.avg_latency_in_sec = (metrics.avg_latency_in_sec
-                    * metrics.num_latency
-                    + cur_metrics.avg_latency_in_sec * cur_metrics.num_latency)
-                    / (metrics.num_latency + cur_metrics.num_latency);
+                    * (metrics.num_latency as f64)
+                    + cur_metrics.avg_latency_in_sec * (cur_metrics.num_latency as f64))
+                    / (metrics.num_latency + cur_metrics.num_latency) as f64;
```

Comment:
All this casting is a bit hard to follow. We first cast num_latency to f64, then cur_metrics' num_latency, then the combined operation of the both of them. Could we perhaps just cast them once as variables to make reading this statement easier? Right now a misplaced closing parenthesis could mess this whole statement up in a not-obvious way.